### DISCUSSION TITLE
🙋Update van 4.20 naar 5.x

### DISCUSSION THREAD

**Discussion by bennybubble** [dsmrreader/dsmr-reader#1793] [Questions & answers / How-to's] [answered]
### Description

Ik wil mijn huidige versie van DSMR (4.20) update naar de meest recente 5.X versie.
Volgens de [docs](https://dsmr-reader.readthedocs.io/nl/v5/tutorial/upgrading/to-v5.html#upgrade-to-dsmr-reader-v5) moet ik Postgresql updaten (heb nog 9.6.24). Mijn vraag is welke versie moet ik hebben en hoe doe ik dat?
DSMR draait op een RPI3 / Raspbian-stretch.

Alvast bedankt!

### DSMR-reader version

4.20

### DSMR-reader platform

Native (e.g. manual installation)

**Comment by dennissiemensma** [comment, marked as answer]
Bedankt voor je vraag. Ik denk dat je het beste kan overgaan naar PostgreSQL 12+ (DSMR-reader 6.x gaat dat vereisen). Versie 14 zou helemaal mooi zijn, zodat je tot eind 2026 dan ondersteuning hebt:
- https://www.postgresql.org/support/versioning/

----

Het wisselen van database is relatief simpel:
- SQL dump maken oude database 
```
sudo su - dsmr

# Dit kan even duren afhankelijk van de grootte
./manage.py dsmr_backup_create --full
```
- DSMR-reader stoppen
- Oude database stoppen
- SQL dump importeren in nieuwe database
- DSMR-aanpassen de nieuwe database laten gebruiken

Alleen zit het meeste werk in de wisseling van databases, want vermoedelijk moet je ook je OS upgraden voor een nieuwere DB-versie.

----

Je kunt overwegen om de Docker-versie te gebruiken: https://github.com/xirixiz/dsmr-reader-docker#dsmr-datalogger-related
Zowel DSMR-reader als PostgreSQL werken in Docker. Het voordeel is dat alles in losse "containers" draait, ongeacht het OS. Je kunt dan ook vrij makkelijk omhoog qua PostgreSQL-versie.

Docker/containers scheelt eigenlijk een hele hoop installatiestappen. Alleen kost het wel wat tijd/werk om te begrijpen wat het doet en hoe het draait. En het is extra belangrijk om te dubbelchecken dat je database-backups veilig zijn, omdat een container relatief makkelijk verwijderd kan worden, inclusief alle data.

----

Als je het op de "native" manier wilt draaien, zul je denk ik ook nog Raspbian 9 moeten upgraden (is sowieso wel goed idee, op zn tijd). 
Maar dan is het een kwestie van bovenstaande SQL-dump te maken, die op te slaan buiten je Pi (dubbelcheck dat de dump goed lijkt!) en dan je Pi herinstalleren met een nieuwe Raspbian, waar ook een nieuwere versie van PostgreSQL bij zit.
Daarna DSMR-reader herinstalleren en bij de installatiestappen die SQL-dump inlezen in je nieuwe database.
